platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,373 @@
+    reported anywhere until [after the build script has been run](https://github.com/department-of-veterans-affairs/vets-website/blob/master/jenkins/common.groovy#L200)
+    
+#### Deployment
+There are currently two kinds of deployments:
+1. [Partial deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#partial-deploy--static-page-changes-only)
+    - For static pages only
+    - Uses the latest `vets-website` release and static assets
+      - Does not trigger Webpack to build the JS and CSS bundles
+      - Instead,
+        [fetches](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/configure-assets.js#L22-L23)
+        the assets [from S3](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/download-assets.js#L111)
+        - Process is triggered [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/Jenkinsfile.content#L32)
+    - Does **NOT** run any content validation such as accessibility tests or
+      broken link checker
+2. [Full deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#full-deploy-of-vagov-client-app)
+    - For `vets-website` code changes
+    - Creates a new release and deploys it
+    - Fetches the latest Drupal content
+      - Falls back to the cached content in S3 from the last successful deploy
+        (full or partial)
+        <!-- TODO: Verify that either deploy will cache the pages.json and files -->
+    - Runs the accessibility tests and link checker
+
+See [here](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/)
+for more details.
+
+#### Pain points
+- Engineers encounter **confusing build errors** that are due to the mismatch in
+  outdated locally cached content and updated Liquid templates
+- Fetching content from the CMS and building HTML files when developing
+  applications **slows down the local builds** for engineers unnecessarily
+- When their applications have nothing to with the CMS
+- It **requires either SOCKS proxy access** or the `npm run fetch-drupal-cache`
+  command, which isn't obvious
+- **Application deployments and PR builds fail** when content in the CMS is
+  invalid
+  - This includes broken links and accessibility violations
+  - We have a safeguard that mostly works; if we run into a failure using the
+    latest CMS content, we try again with the cached content in S3 from the last
+    successful build
+    - This doesn't work when the GraphQL query has been updated; that
+      invalidates the cache
+    - Errors in liquid templates would potentially break the build and falling
+      back to the cache won't help
+
+### High Level Design
+The current front end build will be split up into two distinct builds:
```

Comment:
To confirm: this is talking just about splitting up the _build_ step, not changing anything w/r/t deployment?

Could you explain in this doc how these two build types map to the partial vs. full deploys?

Reply:
Right, this specific line is talking about the build step. The document as a whole I think has to address the deployments, but that's not this part.

I think I can add some points to the section below to map the new builds to the current deploys. :+1:

Reply:
I added the points I mentioned. Does that resolve this?